source/common/upstream/ring_hash_lb.h

```diff
@@ -20,7 +21,9 @@ namespace Upstream {
  * 2) Per-zone rings and optional zone aware routing (not all applications will want this).
  * 3) Max request fallback to support hot shards (not all applications will want this).
  */
-class RingHashLoadBalancer : public LoadBalancer, Logger::Loggable<Logger::Id::upstream> {
+class RingHashLoadBalancer : public LoadBalancerBase,
+                             public LoadBalancer,
```

Comment:
I love seeing a Googler enjoy the virtues of tasteful multiple inheritance. 😉

Reply:
Haha.  We're good with mutliple.  I think we discourage virtual but I'd used it a bunch in QUIC code